sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileTable.scala

```diff
@@ -74,6 +75,7 @@ abstract class FileTable(
     }
     fileIndex match {
       case _: MetadataLogFileIndex => schema
+      case _ if SQLConf.get.respectUserSchemaNullabilityForFileDataSources => schema
```

Comment:
why changing the V2 only?

Reply:
Other code paths are covered in data frame reader, if flag `LEGACY_RESPECT_NULLABILITY_IN_TEXT_DATASET_CONVERSION` is set to true, then schema nullability will be preserved.